### ISSUE TITLE
[Question] Recaptcha plugin sending URL to provider, he might not see captcha

### ISSUE THREAD

**Issue by gitcro** [berstend/puppeteer-extra#267]
If puppeteer gets served a page with a captcha the recaptcha plugin will send the page url to the provider for solving. The provider however might not get a captcha on that same url... will he still be able to solve my captcha?

Thx!

**Comment by Sesamestrong** [comment]
As far as I know, providers use a dedicated proxy to get that URL 100% of the time. For example, if your page is at `https://www.google.com/recaptcha/api2/demo?query=value&id=348920342`, the providers' proxy will be automatically set to intercept requests to that URL so that they return an empty page with a reCAPTCHA in it. Then the provider will visit the URL and will be served the reCAPTCHA. This ensures that providers can get a reCAPTCHA for all page URLs.

TL;DR yes

**Comment by berstend** [comment]
reCAPTCHA solutions are valid per-site, not per page - this is how the captcha solution providers do it. ;-)
